File: CS/MVVMDemo.Wpf/Modules/ViewModelBase/LoginViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;

namespace MVVMDemo.ViewModelBaseDemo {
    public class LoginViewModel : ViewModelBase {
        public string UserName {
            get { return GetProperty(() => UserName); }
            set { SetProperty(() => UserName, value); }
        }
        public string Status {
            get { return GetProperty(() => Status); }
            private set { SetProperty(() => Status, value); }
        }

        [Command]
        public void Login() {
            Status = "User: " + UserName;
        }
        public bool CanLogin() {
            return !string.IsNullOrEmpty(UserName);
        }
    }
}

File: VB/MVVMDemo.Wpf/Modules/ViewModelBase/LoginViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations

Namespace MVVMDemo.ViewModelBaseDemo
    Public Class LoginViewModel
        Inherits ViewModelBase

        Public Property UserName() As String
            Get
                Return GetProperty(Function() UserName)
            End Get
            Set(ByVal value As String)
                SetProperty(Function() UserName, value)
            End Set
        End Property
        Public Property Status() As String
            Get
                Return GetProperty(Function() Status)
            End Get
            Private Set(ByVal value As String)
                SetProperty(Function() Status, value)
            End Set
        End Property

        <Command>
        Public Sub Login()
            Status = "User: " & UserName
        End Sub
        Public Function CanLogin() As Boolean
            Return Not String.IsNullOrEmpty(UserName)
        End Function
    End Class
End Namespace
